packages/assets-controllers/src/NftController.ts

```diff
@@ -1489,6 +1494,56 @@ export class NftController extends BaseControllerV1<NftConfig, NftState> {
     }
   }
 
+  /**
+   * Refetches NFT metadata and updates the state.
+   *
+   * @param nfts - Array of nfts
+   * @param networkClientId - The networkClientId that can be used to identify the network client to use for this request.
+   * @param userAddress - The current user address
+   */
+  async updateNftMetadata(
+    nfts: Nft[],
+    networkClientId?: NetworkClientId,
+    userAddress = this.config.selectedAddress,
+  ) {
+    const chainId = this.getCorrectChainId({ networkClientId });
+    const nftsWithChecksumAdr = nfts.map((nft: Nft) => {
```

Comment:
Nit: Type annotations should be avoided unless they narrow an inferred type. I'm working on [an entry in the new TypeScript guidelines](https://github.com/MetaMask/contributor-docs/pull/74/files#diff-ae03a9f2eb638a25c1d8d7ba853c78220f1ed44a19ad97a8d780cbc0c00a4260R17-R71) that covers this topic.

```suggestion
    const nftsWithChecksumAdr = nfts.map((nft) => {
```